x-pack/packages/kbn-elastic-assistant/impl/knowledge_base/knowledge_base_settings_management/add_entry_button.tsx

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  EuiButton,
+  EuiIcon,
+  EuiPopover,
+  EuiContextMenuItem,
+  EuiContextMenuPanel,
+} from '@elastic/eui';
+import React, { useCallback, useState } from 'react';
+import * as i18n from './translations';
+
+interface Props {
+  isDocumentAvailable?: boolean;
+  isIndexAvailable?: boolean;
+  onDocumentClicked?: () => void;
+  onIndexClicked?: () => void;
+}
+
+export const AddEntryButton: React.FC<Props> = React.memo(
+  ({
+    isDocumentAvailable = true,
+    isIndexAvailable = true,
+    onDocumentClicked,
+    onIndexClicked,
+  }: Props) => {
+    const [isPopoverOpen, setIsPopoverOpen] = useState(false);
+
+    const closePopover = useCallback(() => setIsPopoverOpen(false), []);
+    const handleIndexClicked = useCallback(() => {
+      closePopover();
+      onIndexClicked?.();
+    }, [closePopover, onIndexClicked]);
+
+    const handleDocumentClicked = useCallback(() => {
+      closePopover();
+      onDocumentClicked?.();
+    }, [closePopover, onDocumentClicked]);
+
+    const onButtonClick = useCallback(() => setIsPopoverOpen((prevState) => !prevState), []);
+
+    const items = [
+      <EuiContextMenuItem
+        aria-label={i18n.INDEX}
+        key={i18n.INDEX}
+        icon="index"
+        onClick={handleIndexClicked}
+        disabled={!isIndexAvailable}
+      >
+        {i18n.INDEX}
+      </EuiContextMenuItem>,
+      <EuiContextMenuItem
+        aria-label={i18n.DOCUMENT}
+        key={i18n.DOCUMENT}
+        icon="document"
+        onClick={handleDocumentClicked}
+        disabled={!isDocumentAvailable}
+      >
+        {i18n.DOCUMENT}
+      </EuiContextMenuItem>,
+    ];
+    return onIndexClicked || onDocumentClicked ? (
+      <EuiPopover
+        button={
+          <EuiButton iconType="arrowDown" iconSide="right" onClick={onButtonClick}>
```

Comment:
Consider disabling the `New` button when the KB hasn't been setup yet to avoid the error toaster in the screenshot below:

![error_creating_kb_entry](https://github.com/user-attachments/assets/3e9ccdb1-577b-4cc1-bb40-b0868472c3da)

I'm also wondering if the `Setup Knowledge Base` CTA positioned at the end of the sentence in the screenshot above should be the same size text as the `Learn more` link, or perhaps be more prominent like it is in the assistant:
 
![assistant_setup_button](https://github.com/user-attachments/assets/49b8888b-899a-4c85-bc3c-685cc1fda202)

Reply:
Will disable the `New` button if not available 👍 

And there's still some pending UI changes around the setup button. I chatted with @bojanasan and we decided to hide the table and show the same button as in the assistant as you recommend here.